src/main/java/com/gotocompany/firehose/sink/grpc/GrpcSinkFactory.java

```diff
@@ -21,24 +29,57 @@
  * using the configurations supplied and invoke {@see #create(Map < String, String > configuration, StatsDClient client)}
  * to obtain the GrpcSink sink implementation.
  */
+@Slf4j
 public class GrpcSinkFactory {
 
+    private static final String PERSIST_CERT_DIRECTORY_PATH = "/usr/local/share/ca-certificates";
+
+    private static File createCertFileFromBase64(String base64Cert) throws IOException {
+        byte[] decodedBytes = Base64.getDecoder().decode(base64Cert);
+        File certDirectory = new File(PERSIST_CERT_DIRECTORY_PATH);
+        if (!certDirectory.exists()) {
+            certDirectory.mkdirs();
+        }
+        File certFile = Files.createTempFile(certDirectory.toPath(), "client-cert", ".crt").toFile();
+        log.info("Created certificate file at {}", certFile.getAbsolutePath());
+        try (FileOutputStream fos = new FileOutputStream(certFile)) {
```

Comment:
As we are not handling exception within the block, check if this `try` is required.